clippy_lints/src/len_zero.rs

```diff
@@ -184,34 +184,35 @@ fn check_len_zero(cx: &LateContext, span: Span, name: &Name, args: &[P<Expr>], l
 /// Check if this type has an `is_empty` method.
 fn has_is_empty(cx: &LateContext, expr: &Expr) -> bool {
     /// Get an `ImplOrTraitItem` and return true if it matches `is_empty(self)`.
-    fn is_is_empty(cx: &LateContext, id: &ImplOrTraitItemId) -> bool {
-        if let MethodTraitItemId(def_id) = *id {
-            if let ty::MethodTraitItem(ref method) = cx.tcx.impl_or_trait_item(def_id) {
-                method.name.as_str() == "is_empty" && method.fty.sig.skip_binder().inputs.len() == 1
-            } else {
-                false
-            }
+    fn is_is_empty(id: &ImplOrTraitItem) -> bool {
+        println!("1: {:?}", id);
+        if let ty::MethodTraitItem(ref method) = *id {
+            method.name.as_str() == "is_empty" && method.fty.sig.skip_binder().inputs.len() == 1
         } else {
             false
         }
     }
 
     /// Check the inherent impl's items for an `is_empty(self)` method.
     fn has_is_empty_impl(cx: &LateContext, id: &DefId) -> bool {
-        let impl_items = cx.tcx.impl_items.borrow();
+        let impl_items = cx.tcx.impl_or_trait_items.borrow();
         cx.tcx.inherent_impls.borrow().get(id).map_or(false, |ids| {
-            ids.iter().any(|iid| impl_items.get(iid).map_or(false, |iids| iids.iter().any(|i| is_is_empty(cx, i))))
+            println!("2: {:?}", ids);
+            ids.iter().any(|impl_id| {
+                impl_items.get(impl_id).map_or(false, |i| is_is_empty(i))
+            })
```

Comment:
You should `tcx.impl_or_trait_item(impl_id)` so it can load it from an extern crate - unless this is meant to work only on local impls.